### ISSUE TITLE
returnAfterHttpError false positives

### ISSUE THREAD

**Issue by armsnyder** [go-critic/go-critic#1449]
The `returnAfterHttpError` checker reports an issue even if the `http.Error` call is the last statement in the function. This goes against the intent of the checker.

Code with false positive:

```go
func _(w http.ResponseWriter, err error) {
	if err != nil {
		http.Error(w, "err", 503)
	}
}
```

I am working on a fix for this and will open a PR shortly.
